src/core/timer.c

```diff
@@ -112,7 +124,28 @@ static int timer_add_default_dependencies(Timer *t) {
                 }
         }
 
-        return unit_add_two_dependencies_by_name(UNIT(t), UNIT_BEFORE, UNIT_CONFLICTS, SPECIAL_SHUTDOWN_TARGET, true, UNIT_DEPENDENCY_DEFAULT);
+        if (!UNIT(t)->survive_soft_reboot)
+                return unit_add_two_dependencies_by_name(
+                                UNIT(t),
+                                UNIT_BEFORE, UNIT_CONFLICTS,
+                                SPECIAL_SHUTDOWN_TARGET, true,
+                                UNIT_DEPENDENCY_DEFAULT);
+
+        STRV_FOREACH(target, STRV_MAKE(SPECIAL_REBOOT_TARGET,
+                                       SPECIAL_KEXEC_TARGET,
+                                       SPECIAL_HALT_TARGET,
+                                       SPECIAL_POWEROFF_TARGET)) {
+                r = unit_add_two_dependencies_by_name(UNIT(t),
+                                UNIT_BEFORE,
+                                UNIT_CONFLICTS,
+                                *target,
+                                true,
+                                UNIT_DEPENDENCY_DEFAULT);
+                if (r < 0)
+                        return r;
+        }
```

Comment:
The verbosity of this patch looks terrible. At least the iteration over the targets should be extracted into some helper.

Reply:
true, but it's a trait inherited by the existing code

Reply:
should be better to read now, ptal